strawberry/fastapi/router.py

```diff
@@ -52,26 +62,28 @@ async def __get_root_value():
 
     @staticmethod
     def __get_context_getter(
-        custom_getter: Callable[..., Optional[CustomContext]]
-    ) -> Callable[..., CustomContext]:
-        def dependency(
+        custom_getter: Callable[
+            ..., Union[Optional[CustomContext], Awaitable[Optional[CustomContext]]]
+        ]
+    ) -> Callable[..., Awaitable[CustomContext]]:
+        async def dependency(
             custom_context: Optional[CustomContext],
             background_tasks: BackgroundTasks,
-            request: Request = None,
+            connection: HTTPConnection,
             response: Response = None,
-            ws: WebSocket = None,
         ) -> MergedContext:
-            default_context = {
-                "request": request or ws,
-                "background_tasks": background_tasks,
-                "response": response,
-            }
+            request = cast(Union[Request, WebSocket], connection)
             if isinstance(custom_context, BaseContext):
-                custom_context.request = request or ws
+                custom_context.request = request
```

Comment:
mmh, any reason why we changed this?

Reply:
Yes, see above.  There is a dependency, HTTPConnection, which provides one or the other, avoiding the overhead of having two separate dependencies and choose between them (FastAPI dependencies aren't completely free)

Reply:
Besides, the parameter spec: `request: Request = None` would fail type checking..  Unfortunately, FastApi does not allow `Optional` or `Union` in its dependency type spec, but that is fine, because the only place you'd want to use that is precicely in this case, when getting either a `Request` or a `WebSocket`.